# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavutil/timecode.h
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavutil/timecode.h
--- a/media/ffvpx/libavutil/timecode.h	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavutil/timecode.h	Thu Feb 08 10:39:54 2024 +0000
@@ -98,8 +98,8 @@
 /**
  * Load timecode string in buf.
  *
+ * @param tc       timecode data correctly initialized
  * @param buf      destination buffer, must be at least AV_TIMECODE_STR_SIZE long
- * @param tc       timecode data correctly initialized
  * @param framenum frame number
  * @return         the buf parameter
  *
@@ -149,13 +149,13 @@
 /**
  * Init a timecode struct with the passed parameters.
  *
- * @param log_ctx     a pointer to an arbitrary struct of which the first field
- *                    is a pointer to an AVClass struct (used for av_log)
  * @param tc          pointer to an allocated AVTimecode
  * @param rate        frame rate in rational form
  * @param flags       miscellaneous flags such as drop frame, +24 hours, ...
  *                    (see AVTimecodeFlag)
  * @param frame_start the first frame number
+ * @param log_ctx     a pointer to an arbitrary struct of which the first field
+ *                    is a pointer to an AVClass struct (used for av_log)
  * @return            0 on success, AVERROR otherwise
  */
 int av_timecode_init(AVTimecode *tc, AVRational rate, int flags, int frame_start, void *log_ctx);
@@ -163,8 +163,6 @@
 /**
  * Init a timecode struct from the passed timecode components.
  *
- * @param log_ctx     a pointer to an arbitrary struct of which the first field
- *                    is a pointer to an AVClass struct (used for av_log)
  * @param tc          pointer to an allocated AVTimecode
  * @param rate        frame rate in rational form
  * @param flags       miscellaneous flags such as drop frame, +24 hours, ...
@@ -173,6 +171,8 @@
  * @param mm          minutes
  * @param ss          seconds
  * @param ff          frames
+ * @param log_ctx     a pointer to an arbitrary struct of which the first field
+ *                    is a pointer to an AVClass struct (used for av_log)
  * @return            0 on success, AVERROR otherwise
  */
 int av_timecode_init_from_components(AVTimecode *tc, AVRational rate, int flags, int hh, int mm, int ss, int ff, void *log_ctx);
@@ -180,11 +180,11 @@
 /**
  * Parse timecode representation (hh:mm:ss[:;.]ff).
  *
- * @param log_ctx a pointer to an arbitrary struct of which the first field is a
- *                pointer to an AVClass struct (used for av_log).
  * @param tc      pointer to an allocated AVTimecode
  * @param rate    frame rate in rational form
  * @param str     timecode string which will determine the frame start
+ * @param log_ctx a pointer to an arbitrary struct of which the first field is a
+ *                pointer to an AVClass struct (used for av_log).
  * @return        0 on success, AVERROR otherwise
  */
 int av_timecode_init_from_string(AVTimecode *tc, AVRational rate, const char *str, void *log_ctx);